### PR TITLE
Alerting: Indicate panels without identifier

### PR DESCRIPTION
**Pull request by gillesdemey**
**What is this feature?**

Panel IDs are optional. This wasn't taken in to account in the `DashboardPicker` and when opening the selector and choosing a dashboard that had any panels without `id` the front-end would crash.

The typings were both incorrect and also not correctly applied (`dashboardDTO.panels` is typed as `any` 😖)

This change shows the panels without identifiers but disables it from being selected and adds an indicator that those cannot be selected.

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/DashboardPicker.tsx [hunk 1]
```diff
@@ -73,11 +74,12 @@ export const DashboardPicker = ({ dashboardUid, panelId, isOpen, onChange, onDis
 
   const filteredPanels =
     dashboardResult?.dashboard?.panels
-      ?.filter((panel): panel is PanelDTO => typeof panel.id === 'number' && typeof panel.type === 'string')
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/rule-editor/DashboardPicker.tsx:76]: I removed the filter here, omitting the panels from the selection seemed confusing for the user

**Review by VikaCep — APPROVED** [review]
The solution works great!

I'm just wondering, does it make sense that a panel ID is optional though? Why wouldn't a panel have an ID?

**Comment by gillesdemey** [comment]
> I'm just wondering, does it make sense that a panel ID is optional though? Why wouldn't a panel have an ID?

It doesn't to me, no. However it seems like the API does allow dashboards to be created with panels that have no identifiers so we just have to roll with it :D

**Review by konrad147 — APPROVED** [review]
LGTM!

**Review by soniaAguilarPeiron — APPROVED** [review]
💯

**Pull request merged**